lib/spack/spack/cmd/checksum.py

```diff
@@ -83,22 +83,17 @@ def checksum(parser, args):
     pkg = pkg_cls(spack.spec.Spec(args.package))
 
     url_dict = {}
-    versions = args.versions
-    if (not versions) and args.preferred:
+    if not args.versions and args.preferred:
         versions = [preferred_version(pkg)]
+    else:
+        versions = list(map(Version, args.versions))
```

Comment:
```suggestion
        versions = [Version(v) for v in args.versions]
```

I just find this more readable even though this isn't an awful place to use `map`.